common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -885,3 +878,40 @@ def get_username(strategy, details, backend, user=None, *args, **kwargs):
     else:
         final_username = storage.user.get_username(user)
     return {'username': final_username}
+
+
+def set_learner_active_enterprise(user=None, backend=None, *args, **kwargs):
+    """
+    Make 'active' a user's enterprise,
+    if the currently 'active' enterprise in EnterpriseCustomerUser does not match the SAML idp-enterprise
+    """
+    from openedx.features.enterprise_support.api import (
+        EnterpriseApiClient,
+        EnterpriseApiException,
+        get_enterprise_learner_data,
+    )
+    if is_provider_saml(backend.name, kwargs):
+        saml_prefix = 'saml-'
+        idp_name = kwargs['response']['idp_name']
+        idp_name = saml_prefix + idp_name
+        enterprise_learner_data = get_enterprise_learner_data(user)
```

Comment:
we can not use this method at this point as the user in the request is anonymous. However, we can check the session and get the enterprise_customer from there in case it is present.